sources/Scripts/selfReferencingLinks.py

```diff
@@ -14,21 +14,31 @@ def linkSelfReferences():
     listOfTosafotRefWithSource = findEverySelfReference(tosRefs)
     listOfRashiLinks = createLinks(listOfRashiRefsWithSource)
     listOfTosafotLinks = createLinks(listOfTosafotRefWithSource)
-    functions.post_link(listOfRashiLinks)
-    functions.post_link(listOfTosafotLinks)
 
-    print(listOfRashiLinks)
-    print(listOfTosafotLinks)
-    # print(len(listOfRashiLinks))
-    # print(len(listOfTosafotLinks))
+
+    brokenLinks = open('C:\Users\joshuaegoldmeier\Documents\BrokenLinks.txt','a')
```

Comment:
I would take another approach. Use the Ref.is_ref() function to check if the string is a valid ref. except Exception is one of the most dangerous lines you can write in python, as it will handle everything, even keyboard interrupts.